overlord/devicestate/devicestate_test.go

```diff
@@ -330,12 +330,29 @@ func (s *deviceMgrSuite) TestDeviceManagerEnsureSeededHappy(c *C) {
 }
 
 func (s *deviceMgrSuite) TestDeviceManagerEnsureBootOkSkippedOnClassic(c *C) {
+	s.bootloader.GetErr = fmt.Errorf("should not be called")
 	release.OnClassic = true
 
 	err := devicestate.EnsureBootOk(s.mgr)
 	c.Assert(err, IsNil)
 }
 
+func (s *deviceMgrSuite) TestDeviceManagerEnsureBootOkSkippedOnNonRunModes(c *C) {
+	s.bootloader.GetErr = fmt.Errorf("should not be called")
+	devicestate.SetOperatingMode(s.mgr, "install")
+
```

Comment:
Maybe we should explicitly set `release.MockOnClassic(false)` to prevent fallling in the classic test condition?

Reply:
that's done for all the tests here in SetUpTest